domain/snyk/persistence/git_persistence_provider.go

```diff
@@ -256,8 +261,54 @@ func (g *GitPersistenceProvider) Exists(folderPath, commitHash string, p product
 	return false
 }
 
+func (g *GitPersistenceProvider) fileSchema(fullPath string) (string, hashedFolderPath, string, product.Product, error) {
+	// file name structure is schemaVersion.hashedFolderPath.commitHash.productName.json
+	s := strings.Split(filepath.Base(fullPath), ".")
+	if len(s) != 5 {
+		return "", "", "", "", fmt.Errorf("failed to parse file name %s", fullPath)
+	}
+	schemaVersion := s[0]
+	hash := hashedFolderPath(s[1])
+	commitHash := s[2]
+	p := product.ToProduct(s[3])
+	return schemaVersion, hash, commitHash, p, nil
+}
+
+func (g *GitPersistenceProvider) deleteCacheEntryIfExpired(fullPath string) error {
+	g.logger.Debug().Msgf("deleting cached scan file %s", fullPath)
+	schemaVersion, hash, commitHash, p, _ := g.fileSchema(fullPath)
+	if schemaVersion != SchemaVersion {
+		return g.deleteFromDiskAndCache(fullPath, hash, commitHash, p)
+	}
+
+	// Check last modified date
+	fileInfo, err := os.Stat(fullPath)
+	if err != nil {
+		return err
+	}
+	// If elapsed time is > ExpirationInSeconds, delete the file
+	if time.Since(fileInfo.ModTime()) > time.Duration(ExpirationInSeconds)*time.Second {
+		return g.deleteFromDiskAndCache(fullPath, hash, commitHash, p)
+	}
+
+	return nil
+}
+
+func (g *GitPersistenceProvider) deleteFromDiskAndCache(fullPath string, hash hashedFolderPath, commitHash string, p product.Product) error {
+	// if file has incorrect schema we just delete it
+	err := os.Remove(fullPath)
```

Comment:
just wondering, what happens with a windows file lock?

Reply:
I think we will just need to wait for the next 12 hrs. Maybe I will need to adjust the init logic to ensure validity before loading the file into cache.